src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,670 @@
+      return parentID + '-' + node.getAttribute('class') + '-' + extraClasses;
+    }
+
+    function computeCachedCSSStyles(node, className, cacheKey, properties) {
+      var timings = gcsLookup.get(cacheKey);
+
+      if (!timings) {
+        timings = computeCSSStyles($window, node, properties);
+        if (timings.animationIterationCount) {
+          timings.animationIterationCount = parseInt(timings.animationIterationCount || 1, BASE_TEN);
+        }
+      }
+
+      // we keep putting this in multiple times even though the value and the cacheKey are the same
+      // because we're keeping an interal tally of how many duplicate animations are detected.
+      gcsLookup.put(cacheKey, timings);
+      return timings;
+    }
+
+    function computeCachedCSSStaggerStyles(node, className, cacheKey, properties) {
+      var stagger = {};
+      var isRepeated = gcsLookup.count(cacheKey) > 0;
+
+      if (isRepeated) {
+        var staggerCacheKey = cacheKey + '-stagger';
+        stagger = gcsStaggerLookup.get(staggerCacheKey);
+
+        if (!stagger) {
+          var staggerClassName = pendClasses(className, '-stagger');
+
+          $$jqLite.addClass(node, staggerClassName);
+
+          stagger = computeCSSStyles($window, node, properties);
+
+          $$jqLite.removeClass(node, staggerClassName);
+
+          gcsStaggerLookup.put(staggerCacheKey, stagger);
+        }
+      }
+
+      return stagger;
+    }
+
+    var cancelLastRafRequest, rafDeferred, bod = $document[0].body;
+    function waitUntilQuiet() {
+      if (cancelLastRafRequest) {
+        cancelLastRafRequest(); //cancels the request
+      }
+      if (!rafDeferred) {
+        rafDeferred = $qRaf.defer();
+      }
+      cancelLastRafRequest = $$rAF(function() {
+        gcsLookup.flush();
+        gcsStaggerLookup.flush();
+
+        //the line below will force the browser to perform a repaint so
+        //that all the animated elements within the animation frame will
+        //be properly updated and drawn on screen. This is required to
+        //ensure that the the preparation animation is properly flushed so that
+        //the active state picks up from there. DO NOT REMOVE THIS LINE.
+        var a = bod.offsetWidth + 1;
+
+        var deferred = rafDeferred;
+        rafDeferred = null;
+        deferred.resolve();
+      });
+      return rafDeferred.promise;
+    }
+
+    return init;
+
+    function computeTimings(node, className, cacheKey) {
+      var timings = computeCachedCSSStyles(node, className, cacheKey, {
+        transitionDuration:      TRANSITION_PROP + DURATION_KEY,
+        transitionDelay:         TRANSITION_PROP + DELAY_KEY,
+        transitionProperty:      TRANSITION_PROP + PROPERTY_KEY,
+        animationDuration:       ANIMATION_PROP  + DURATION_KEY,
+        animationDelay:          ANIMATION_PROP  + DELAY_KEY,
+        animationIterationCount: ANIMATION_PROP  + ANIMATION_ITERATION_COUNT_KEY
+      });
```

Comment:
Yes good point. Maybe have it declared at the top of the file where all the other constants are defined.